### PR TITLE
Add google account authentication (#41)

### PR DESCRIPTION
**Pull request by bbjubjub2494**
This is more so geared for a demo.

IMPORTANT: the app now won't build without the passphrase, see discord for instructions.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -14,7 +14,8 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.HumanLearning">
-        <activity android:name=".DataCreationActivity"/>
+        <activity android:name=".ui.login.GoogleSignInActivity"></activity>
```
Comment (Nitwix) [inline app/src/main/AndroidManifest.xml]: does the 'ui' folder have a different purpose than 'view'?

**Review by Nitwix — COMMENTED** [review]
Just a question about project hierarchy

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (bbjubjub2494) [reply app/src/main/AndroidManifest.xml]: That wasn't my decision, but we're not even using the code that generated this anymore so we might as well name this whatever. Should it be in the top-level like the other activities in your book?

**Review by wattsecs — CHANGES_REQUESTED** [review]
Implementation looks good to me. I would only suggest putting the strings into resources.

**Comment by bbjubjub2494** [comment]
> Implementation looks good to me. I would only suggest putting the strings into resources.

Should be addressed lmk

**Comment by bbjubjub2494** [comment]
ok time for CI rain dance

**Review by wattsecs — APPROVED** [review]
My comments have been addressed

**Review by Nitwix — APPROVED** [review]
All good for me!

**Pull request merged**